### PR TITLE
Migrate Cargo.toml testing to GH Actions

### PR DESCRIPTION
**Pull request by mbestavros**
This allows us to remove the lengthy `toml-cli` installation from Travis.

Relevant tests have been moved from `.hooks/pre-push.d` to a new `.tests` directory.

Related: #246

### CODE REVIEW

**Comment by mbestavros** [comment]
For reference, [here's](https://github.com/enarx/enarx/runs/453412239?check_suite_focus=true) an example of a license field test failure.

File: .github/workflows/enarx-standard.yml [hunk 1]
```diff
@@ -4,9 +4,25 @@ name: Enarx Standard Tests
 on: [pull_request]
 
 jobs:
+  # Ensure the testing infrastructure is working as intended.
   sanity-check:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@master
+    - uses: actions/checkout@v1
```
Comment (npmccallum) [inline .github/workflows/enarx-standard.yml]: If there's a reason to change this, it should be an independent logical commit with clear explanation.

File: .github/workflows/enarx-standard.yml [hunk 2]
```diff
@@ -4,9 +4,25 @@ name: Enarx Standard Tests
...
+  # Enforce correct fields in Cargo.toml files.
+  cargo-toml-tests:
+    runs-on: ubuntu-latest
+    container: quay.io/enarx/fedora-test
+    steps:
+    - uses: actions/checkout@v1
+    
+    # Ensure Cargo edition is correct.
+    - name: Cargo edition
+      run: .tests/cargo-edition-test
```
Comment (npmccallum) [inline .github/workflows/enarx-standard.yml]:
```suggestion
      run: .tests/cargo-edition
```

File: .github/workflows/enarx-standard.yml [hunk 3]
```diff
@@ -4,9 +4,25 @@ name: Enarx Standard Tests
...
+
+    # Ensure Cargo.toml's license field is Apache 2.0.
+    - name: License field
+      run: .tests/toml-license-field-test
```
Comment (npmccallum) [inline .github/workflows/enarx-standard.yml]:
```suggestion
      run: .tests/toml-license-field
```

File: .github/workflows/enarx-standard.yml (hunk 1 above)
Comment (mbestavros) [reply .github/workflows/enarx-standard.yml]: You caught me realizing this was a tagging system in real time :stuck_out_tongue:

**Pull request merged**